lib/assert.js

```diff
@@ -369,9 +382,159 @@ function isSpecial(obj) {
 }
 
 const typesToCallDeepStrictEqualWith = [
-  isKeyObject, isWeakSet, isWeakMap, Buffer.isBuffer,
+  isKeyObject, isWeakSet, isWeakMap, Buffer.isBuffer, isSharedArrayBuffer,
 ];
 
+function compareMaps(actual, expected, comparedObjects) {
+  if (actual.size !== expected.size) {
+    return false;
+  }
+  const safeIterator = FunctionPrototypeCall(SafeMap.prototype[SymbolIterator], actual);
+
+  comparedObjects ??= new SafeWeakSet();
+
+  for (const { 0: key, 1: val } of safeIterator) {
+    if (!MapPrototypeHas(expected, key)) {
+      return false;
+    }
+    if (!compareBranch(val, MapPrototypeGet(expected, key), comparedObjects)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function compareArrayBuffers(actual, expected) {
```

Comment:
Should we rename the function to make it clearer it accepts any `actual` that is a superset of `expected`? The `compareMaps` function just above only accepts `Map` that are deep strict equal, so that's a bit confusing.
```suggestion
function comparePartiallyArrayBuffers(actual, expected) {
```